From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Wed, 14 Aug 2013 10:27:05 +0200
Subject: [PATCH] configure: if using ssl link against libcrypto, too

otherwise linking fails with:
arm-v5te-linux-gnueabi-gcc -o ssmtp ssmtp.o arpadate.o base64.o xgethostname.o md5auth/md5c.o md5auth/hmac_md5.o  -lnsl  -lssl
/opt/OSELAS.Toolchain-2012.12.1/arm-v5te-linux-gnueabi/gcc-4.7.2-glibc-2.16.0-binutils-2.22-kernel-3.6-sanitized/lib/gcc/arm-v5te-linux-gnueabi/4.7.2/../../../../arm-v5te-linux-gnueabi/bin/ld:
ssmtp.o: undefined reference to symbol 'X509_free@@OPENSSL_1.0.0'
/opt/OSELAS.Toolchain-2012.12.1/arm-v5te-linux-gnueabi/gcc-4.7.2-glibc-2.16.0-binutils-2.22-kernel-3.6-sanitized/lib/gcc/arm-v5te-linux-gnueabi/4.7.2/../../../../arm-v5te-linux-gnueabi/bin/ld:
note: 'X509_free@@OPENSSL_1.0.0' is defined in DSO
XXXXX/sysroot-target/usr/lib/libcrypto.so.1.0.0 so try adding it to the
linker command line
XXXXX/sysroot-target/usr/lib/libcrypto.so.1.0.0: could not read symbols:
Invalid operation
collect2: error: ld returned 1 exit status

Reported-by: Alexander Dahl <post@lespocky.de>
Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 2d7221d..0633add 100644
--- a/configure.in
+++ b/configure.in
@@ -52,7 +52,7 @@ AC_ARG_ENABLE(ssl,
 [  --enable-ssl           support for secure connection to mail server])
 if test x$enableval = xyes ; then
 	AC_DEFINE(HAVE_SSL)
-	LIBS="$LIBS -lssl"
+	LIBS="$LIBS -lssl -lcrypto"
 fi
 enableval=""
 
